### PR TITLE
Add proposal Steering Committee application, terms, and role

### PR DESCRIPTION
**Pull request by joshuagl**
This proposal clarifies the SLSA Steering Committee, specifically with regards to:

* responsibilities
* difference to core maintainer responsibilities
* application, nomination and terms

### CODE REVIEW

File: 0007/README.md
Comment (MarkLodato) [inline 0007/README.md]: Suggestion from @brunodom was to soften this language. Maybe just say that:
- We expect SC Members to be well versed in the project (or to show expertise, or something like that)
- In practice SC Members are likely also Core Maintainers, but this is not a requirement.
- Remove the sentence about being a stepping stone.

File: 0007/README.md
Comment (MarkLodato) [inline 0007/README.md]:
```suggestion
The original SLSA Steering Committee (SC) has passed its two year term and, per
```

So make it clear what "SC" means in this doc.

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md]: [aff2902](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/aff29026d276b7dbb8659f75224ecf80cc3203ae) softens the language, as suggested.

@brunodom @MarkLodato PTAL.

**Comment by kimsterv** [comment]
+1 Lgtm!

File: 0007/README.md
Comment (brunodom) [reply 0007/README.md]: LGTM.

**Review by MarkLodato — APPROVED** [review]
@brunodom and @kimsterv could you approve via the GitHub interface (Files Changed -> Review changes -> Approve) so that others can easily see the checkmark in the upper right? Thanks!

**Review by kimsterv — APPROVED** [review]

**Review by trishankatdatadog — APPROVED** [review]

File: 0007/README.md
Comment (melba-lopez) [inline 0007/README.md:75]: where do they submit their nomination? and what are the timelines so people know ahead of time.  (i.e. vacations/life happens)

File: 0007/README.md
Comment (melba-lopez) [inline 0007/README.md:84]: Shouldn't we allow the community to vote as well for the SC Members or maintainers seat?  SC can have the majority vote, but if you don't have community consensus, this leads to potential bias.

File: 0007/README.md
Comment (melba-lopez) [inline 0007/README.md:92]: When is the actual timeline?  every year on October... as an example.

File: 0007/README.md
Comment (melba-lopez) [inline 0007/README.md:198]: thanks for adding this.

File: 0007/README.md
Comment (melba-lopez) [inline 0007/README.md:182]: may want to add a cadence.. quarterly.. every 6 months..etc.

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md:75]: Good questions. I expect we will figure thesse details out as part of the implementation (a PR against [slsa-framework/governance](https://github.com/slsa-framework/governance/)), but my initial suggestion would be:
* nominations are GitHub Issues
* timeline/cadence would be documented in the governance repository and the nomination period would be broadly communicated (in Slack, email, meetings, etc)

File: 0007/README.md
Comment (abacchi) [reply 0007/README.md:92]: For maintiners maybe it should be an open door period rather than one specific timeframe. SC member applicaiton should be a fixed annual application timeline though.

File: 0007/README.md
Comment (abacchi) [inline 0007/README.md:123]: It may be nice for the SC group to present/review their activities quarterly / 6 month'ly at the standard SLSA meetings to improve transparency as well.

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md:92]: Actual timeline tbd. I believe we should test the proposed change to the governance repo before we fully accept it, by running a nomination cycle for the SC. Then will do things annually from that (approximate) date.

Agreed that for maintainers, it should not be a specific timeframe. Good point!

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md:182]: Good call, I'll add a note that we should do this regularly and at a minimum following each SC nomination cycle.

Comment (joshuagl) [reply 0007/README.md:182]: Added in https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4d91b1ab5415d562e4c3560d3edbbba923070d79

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md:123]: I like the suggestion and agree with the principal. I am wary of codifying a periodicity, but added some text in [6a96a8c](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/6a96a8cc37d3fe4d33321cacd63d49a58d6aae5e). WDYT?

**Review by joshuagl — COMMENTED** [review]
Thanks all for the reviews and feedback so far. I've pushed some minor changes in response to review feedback:

* noted that we should perform purging of unneccessary GitHub permissions on a regular cadence
* added SC Member responsibilities:
   * sharing relevant SCI WG happeninings with the community (making this sharing 2-way)
   * that the SC should share updates in the community meeting (I've added a section to the meeting minute template in the Google doc too)

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md:84]: We have been trying to avoid a formal voting process, as they can be onerous for the people running them. That also helps us avoid the question of how to identify elligible voters.

I hope that the public nomination process will help keep people honest. If we are to use GitHub Issues for nominations, that would allow folks to express support (though admittedly in a non-anonymous way).

Comment (MarkLodato) [reply 0007/README.md:84]: IMO voting is overkill for such a small community. If the community grows 10x in size, then maybe we reconsider?

File: 0007/README.md
Comment (MarkLodato) [reply 0007/README.md:92]: My feeling was rolling, as @abacchi said. We could have some sort of script that sends a notification when a SC member or maintainer's approval is out of date.

File: 0007/README.md
Comment (lehors) [reply 0007/README.md:75]: I think it'd be better to have nominations made in the form of a PR that changes the list of maintainers or SC members as applicable. The PR description can be used to provide whatever information is required and this enforces the maintenance of the list in a publicly visible manner.

File: 0007/README.md
Comment (lehors) [reply 0007/README.md:84]: The biggest challenge in holding such a vote is that it raises the question of who gets to vote. For this reason, I think it is better to keep the decision to the current maintainers and SC members even though it is prone to conflicts of interest.

File: 0007/README.md
Comment (lehors) [inline 0007/README.md]: For the sake of completeness this should include the pro:
```suggestion
*   Dissolve the SLSA Steering Committee in favor of steering from the SCI WG
    *   Pro: simplified governance structure
    *   Con: Strategy may be driven by those without deep familiarity with
        the project.
    *   Con: SLSA community may not feel an appropriate sense of ownership and
        autonomy.
```

File: 0007/README.md
Comment (lehors) [inline 0007/README.md]: Err... I think you're falling into subjectivity land on this one. :-)

**Review by lehors — COMMENTED** [review]
Thanks for putting this together. I appreciate the candid approach about the current state we are in and think this is going in the right direction.
I added a few minor comments and have one nit:
The document uses both "OSSF" and "OpenSSF". I know OpenSSF is not consistent on this (we have openssf.org and github.com/ossf) but I think we should stick to OpenSSF which seems to be the dominant form.

**Comment by brunodom** [comment]
@MarkLodato, unfortunately, I can't approve it now because I am traveling
and forgot my security key. I am not able to set up the Authenticator
without logging into my account. Basically, I am lockout these days.
However, consider it approved on my side.

Best Regards!
-Bruno Domingues

File: 0007/README.md
Comment (abacchi) [reply 0007/README.md:123]: yes that update looks great.

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md]: Agreed, added in [4432b4a](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4432b4a8dd41a8ee2ab73f1a074b409ab7e4e07e). Thanks!

File: 0007/README.md
Comment (joshuagl) [reply 0007/README.md]: Fair comment, I've removed this in [4432b4a](https://github.com/slsa-framework/slsa-proposals/pull/15/commits/4432b4a8dd41a8ee2ab73f1a074b409ab7e4e07e).

**Comment by bobcallaway** [comment]
LGTM as well

**Comment by MarkLodato** [comment]
Friendly ping. What do we need to do to get this over the line and accepted?

**Review by lehors — APPROVED** [review]

**Comment by lehors** [comment]
> Friendly ping. What do we need to do to get this over the line and accepted?

Who are you pinging and asking this to @MarkLodato ? For what it's worth, I approve. But I think if the current SC approves that should do it since technically the SC owns the governance from what I understand. If you want to be more open about it I suggest you raise it as a proposal to be approved on the next spec group call.

**Comment by trishankatdatadog** [comment]
I would prefer that we get the entire current SC to approve this.

**Comment by mlieberman85** [comment]
I will double check and approve shortly. I've read it, but want to make sure I haven't missed anything.

**Comment by joshuagl** [comment]
It would be good to have approval from the full, current, @slsa-framework/slsa-steering-committee for this proposal. 

- @MarkLodato, @kimsterv and @trishankatdatadog have approved the PR
- @brunodom [approved in a comment]9https://github.com/slsa-framework/slsa-proposals/pull/15#issuecomment-1667159496)
- we're still waiting for @david-a-wheeler and @mlieberman85

**Review by mlieberman85 — APPROVED** [review]
LGTM

**Review by brunodom — APPROVED** [review]

File: 0007/README.md
Comment (david-a-wheeler) [reply 0007/README.md:75]: You really want an *odd* number when voting, or you risk deadlocks.

File: 0007/README.md
Comment (david-a-wheeler) [inline 0007/README.md:158]: I doubt you want that many people as owners. Owners can delete everything.

File: 0007/README.md
Comment (david-a-wheeler) [reply 0007/README.md:84]: I think it's important to have a mechanism for making hard decisions (voting), and then do your best to avoid using it (e.g., by determining why there are differences and working out compromises that address all major concerns).

File: 0007/README.md
Comment (david-a-wheeler) [reply 0007/README.md:123]: We want at least 2 different organizations represented, yes? Shouldn't that be clearly stated?

**Review by david-a-wheeler — APPROVED** [review]
It looks okay to me with the following issues:

1. We need to ensure it stays multi-organizational. I believe the current rule is "no more than 2 from 1 organization, just like OpenSSF"; that should continue.
2. Let's ensure there is no deadlock in the voting process. The simple approach is to have an odd number of people; another alternative is to have a tie-breaking rule. But if we have that much contention and *must* make a decision, then we need to have a way to break ties.

**Comment by joshuagl** [comment]
Thanks for the review.

> It looks okay to me with the following issues:
> 
>     1. We need to ensure it stays multi-organizational. I believe the current rule is "no more than 2 from 1 organization, just like OpenSSF"; that should continue.

Completely agree. This requirement is currently part of our documented governance model and shall continue under this proposal.

>     2. Let's ensure there is no deadlock in the voting process. The simple approach is to have an odd number of people; another alternative is to have a tie-breaking rule. But if we have that much contention and _must_ make a decision, then we need to have a way to break ties.

Good call out. The current governance documents require up to 7 participants in the SC. This proposal changes that to "a minimum of 3 and ideally about 5". 
We should explicitly define a tie-breaking, as I think that is more resilient than trying to require an odd number of members of the SC.

**Pull request merged**
